src/frequenz/dispatch/_bg_service.py

```diff
@@ -23,6 +27,77 @@
 """The logger for this module."""
 
 
+class _MergeStrategy(ABC):
+    """Base class for strategies to merge running intervals."""
+
+    @abstractmethod
+    def _get_filter_function(
+        self,
+        scheduler: DispatchScheduler,
+    ) -> Callable[[Dispatch], bool]:
+        """Get a filter function for dispatches.
+
+        Args:
+            scheduler: The dispatch scheduler.
+
+        Returns:
+            A filter function.
+        """
+
+
+class MergeByType(_MergeStrategy):
+    """Merge running intervals based on the dispatch type."""
+
+    def __init__(self) -> None:
+        """Initialize the strategy."""
+        self._scheduler: DispatchScheduler
+        self._new_dispatch: Dispatch
+
+    def _get_filter_function(
+        self, scheduler: DispatchScheduler
+    ) -> Callable[[Dispatch], bool]:
+        """Get a filter function for dispatches."""
+        self._scheduler = scheduler
+        return self._filter_func
+
+    def _criteria(self, dispatch: Dispatch) -> bool:
+        """Define the criteria for checking other running dispatches."""
+        return dispatch.type == self._new_dispatch.type
```

Comment:
I would not save the `_new_dispatch` as an instance attribute, but pass it in the args.

```suggestion
    def _criteria(self, existing_dispatch: Dispatch, new_disaptch: Dispatch) -> bool:
        """Define the criteria for checking other running dispatches."""
        return existing_dispatch.type == new_dispatch.type
```